src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -454,19 +459,21 @@ private static boolean skipDoubleBraceInstInit(Details details) {
      * @return true if block has single-line format and is alone on a line.
      */
     private static boolean isBlockAloneOnSingleLine(Details details) {
-        final DetailAST rcurly = details.rcurly;
-        final DetailAST lcurly = details.lcurly;
         DetailAST nextToken = details.nextToken;
-        while (nextToken.getType() == TokenTypes.LITERAL_ELSE) {
-            nextToken = Details.getNextToken(nextToken);
+        if (nextToken != null) {
+            while (nextToken.getType() == TokenTypes.LITERAL_ELSE) {
+                nextToken = Details.getNextToken(nextToken);
+            }
         }
-        if (nextToken.getType() == TokenTypes.DO_WHILE) {
+
+        if (nextToken != null && nextToken.getType() == TokenTypes.DO_WHILE) {
```

Comment:
Both this if statement and the previous one can be joined since they both have the same null check on the same variable.

Reply:
Done